File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int ans = 0, n = s.size();
        for(int i = 0;i < n;i++) {
            map<int, int> frq;
            int cnt = 0, j = i;
            while (j < n and frq[s[j] - 'a'] == 0) {
                frq[s[j] - 'a']++;
                cnt++;
                j++;
            } 
            ans = max(ans, cnt);
        }
        return ans;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.dart
class Solution {
  int lengthOfLongestSubstring(String s) {
    Set<String> st = {};
    int l = 0, r = 0;
    int n = s.length;
    int ans = 0;

    while (r < n) {
      while (r < n && !st.contains(s[r])) {
        st.add(s[r]);
        r++;
      }
      ans = (ans > (r - l)) ? ans : (r - l);
      st.remove(s[l]);
      l++;
    }

    return ans;
  }
}